docs/_docs/reference/contextual/context-bounds.md

```diff
@@ -4,50 +4,205 @@ title: "Context Bounds"
 
 ```scala
 def maximum[T: Ord](xs: List[T]): T = xs.reduceLeft(max)
 ```
 
-A bound like `: Ord` on a type parameter `T` of a method or class indicates a context parameter `using Ord[T]`. The context parameter(s) generated from context bounds
-are added as follows:
+A bound like `: Ord` on a type parameter `T` of a method or class indicates a context parameter `using Ord[T]`, which is added to the signature of the enclosing method. The generated parameter is called a _witness_ for the context bound.
 
- - If the method parameters end in an implicit parameter list or using clause,
-   context parameters are added in front of that list.
- - Otherwise they are added as a separate parameter clause at the end.
-
-Example:
+For instance the `maximum` method above expands to
+```scala
+def maximum[T](xs: List[T])(using Ord[T]): T = ...
+```
+Context bounds can be combined with subtype bounds. If both are present, subtype bounds come first, e.g.
 
 ```scala
-def f[T: C1 : C2, U: C3](x: T)(using y: U, z: V): R
+def f[T <: B : C](x: T): R = ...
 ```
 
-would expand to
+## Named Context Bounds
 
+A context bound can be given a name with an `as` clause. For example, assume the following trait definitions.
+```scala
+  trait SemiGroup[A]:
+    extension (x: A) def combine(y: A): A
+
+  trait Monoid[A] extends SemiGroup[A]:
+    def unit: A
+```
+We can write `reduce` function over lists of monoid instances like this:
 ```scala
-def f[T, U](x: T)(using _: C1[T], _: C2[T], _: C3[U], y: U, z: V): R
+  def reduce[A: Monoid as m](xs: List[A]): A =
+    xs.foldLeft(m.unit)(_ `combine` _)
 ```
+We use `as x` after the type of a context bound to bind the instance to `x`. This is analogous to import renaming, which also introduces a new name for something that comes before.
 
-Context bounds can be combined with subtype bounds. If both are present, subtype bounds come first, e.g.
+In a context bound with a naming clause the witness parameter carries the given name. For instance the expanded signature of `reduce` would be
+```scala
+  def reduce[A](xs: List[A])(using m: Monoid[A]): A
+```
+Since the context parameter now has a name, it can be referred
+to in the body of `reduce`. An example is the `m.unit` reference in the definition above.
+
+If the context bound does not carry an `as` clause, the generated witness parameter gets a compiler-synthesized name. However, a [currently experimental
+language extension](../experimental/default-names-context-bounds.md) would in this case give the context parameter the same name as the bound type parameter.
+
+Named context bounds were introduced in Scala 3.6.
+
+## Aggregate Context Bounds
 
+A type parameter can have several context bounds. If there are multiple bounds, they are written inside braces `{...}`. Example:
```

Comment:
Perhaps there should be a note saying what happens if one wants to combine a subtype bound with *multiple* context bounds.

Reply:
I think it's fine since the two are orthogonal and we cover combination of subtype and context bounds in line 22 before.